core/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/core/httpapi/HttpExtensions.kt

```diff
@@ -0,0 +1,64 @@
+package com.amazon.opendistroforelasticsearch.alerting.core.httpapi
+
+import com.amazon.opendistroforelasticsearch.alerting.core.model.HttpInput
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.suspendCancellableCoroutine
+import org.apache.http.HttpResponse
+import org.apache.http.client.config.RequestConfig
+import org.apache.http.client.methods.HttpGet
+import org.apache.http.client.utils.URIBuilder
+import org.apache.http.concurrent.FutureCallback
+import org.apache.http.nio.client.HttpAsyncClient
+import org.apache.http.util.EntityUtils
+import org.elasticsearch.common.Strings
+import org.elasticsearch.common.xcontent.LoggingDeprecationHandler
+import org.elasticsearch.common.xcontent.NamedXContentRegistry
+import org.elasticsearch.common.xcontent.XContentType
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+
+suspend fun <C : HttpAsyncClient, T> C.suspendUntil(block: C.(FutureCallback<T>) -> Unit): T =
+        suspendCancellableCoroutine { cont ->
+            block(object : FutureCallback<T> {
+                override fun cancelled() {
+                    cont.resumeWith(Result.failure(CancellationException("Request cancelled")))
```

Comment:
can we use `cont.resumeWithException(CancellationException(...))` here?

Reply:
Will do so, thanks!